### ISSUE TITLE
Windows style line endings not supported

### ISSUE THREAD

**Issue by mmdaminmlz** [wmww/Pinecone#27]
i have compiled pinecone with the make file and i got this error for every file i test.
```
$ ./pinecone ./examples/hello_world.pn
error in './examples/hello_world.pn' on line 1:
'   invalid token '
    print: "Hello World!"
                         ^
error in './examples/hello_world.pn' on line 2:
'   invalid token '

    ^
program not executed due to errors
```
I'm running on WSL Ubuntu 16.04 64-bit
```
$ lsb_release -a
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.2 LTS
Release:        16.04
Codename:       xenial
$ uname -a
Linux DESKTOP-53JBK0N 4.4.0-17134-Microsoft #1-Microsoft Tue Apr 10 18:04:00 PST 2018 x86_64 x86_64 x86_64 GNU/Linux
```
![image](https://user-images.githubusercontent.com/21140798/50721202-c83a0b80-10d0-11e9-9f4d-5295c24f2d9f.png)

**Comment by wmww** [comment]
Ah, I'm willing to be I've tested on windows, but never with Windows line endings.

**Comment by mmdaminmlz** [comment]
oh ok. I'll use my text editor to convert line endings to unix CR

**Comment by mmdaminmlz** [comment]
it worked. thank you.
your compiler isn't stable enough.
```
$ ./pinecone ./examples/hello_world.pn
Hello World!
```

**Comment by wmww** [comment]
> your compiler isn't stable enough.

That may be, but in this instance it gave a semi-useful error message and exited gracefully. I'd call that plenty stable. I've tested once or twice on Windows, but never claimed to properly support anything but Linux (even when it *was* being developed and maintained).

Anyway, this should be fixed with 567d6c36ae898dfc99b746ef39200f6ad2929ab4, and there's also now a unit test for it.

**Comment by mmdaminmlz** [comment]
> > your compiler isn't stable enough.
> 
> That may be, but in this instance it gave a semi-useful error message and exited gracefully. I'd call that plenty stable. I've tested once or twice on Windows, but never claimed to properly support anything but Linux (even when it _was_ being developed and maintained).
> 
> Anyway, this should be fixed with [567d6c3](https://github.com/wmww/Pinecone/commit/567d6c36ae898dfc99b746ef39200f6ad2929ab4), and there's also now a unit test for it.
> 
> Edit: I merged your PR #28 instead, which did exactly the same thing as my commit.

this is a full linux environment running on top of windows nt kernel, not same as mingw or cygwin that are gnu ports to windows. i could download, install and run any deb package from orginal ubuntu repos and install it easily on my WSL.
i don't say your project is bad, buggy or other, i respect you and your project, that's a great work, but it's too clear to ignore chars like '\r' by default, same as '\t' and whitespace.
